File: bench/tasks/09-grade-report/go/main.go
package main

import "fmt"

func main() {
	scores := []int{85, 72, 91, 60, 78, 95, 55, 68, 88, 74}
	total, aGrades, bGrades, passing, failing := 0, 0, 0, 0, 0
	for _, s := range scores {
		total += s
		if s >= 90 {
			aGrades++
		}
		if s >= 80 && s < 90 {
			bGrades++
		}
		if s >= 60 {
			passing++
		}
		if s < 60 {
			failing++
		}
	}
	avg := total / len(scores)
	fmt.Println("Class average: " + fmt.Sprint(avg))
	fmt.Println("A grades: " + fmt.Sprint(aGrades))
	fmt.Println("B grades: " + fmt.Sprint(bGrades))
	fmt.Println("Passing: " + fmt.Sprint(passing))
	fmt.Println("Failing: " + fmt.Sprint(failing))
}

File: bench/tasks/09-grade-report/rust/main.rs
fn main() {
    let scores = [85, 72, 91, 60, 78, 95, 55, 68, 88, 74];
    let total: i32 = scores.iter().sum();
    let avg = total / scores.len() as i32;
    let a_grades = scores.iter().filter(|&&s| s >= 90).count();
    let b_grades = scores.iter().filter(|&&s| s >= 80 && s < 90).count();
    let passing = scores.iter().filter(|&&s| s >= 60).count();
    let failing = scores.iter().filter(|&&s| s < 60).count();
    println!("Class average: {}", avg);
    println!("A grades: {}", a_grades);
    println!("B grades: {}", b_grades);
    println!("Passing: {}", passing);
    println!("Failing: {}", failing);
}
